spec/controllers/users/sessions_controller_spec.rb

```diff
@@ -252,6 +253,7 @@
         sp_request_url_present: false,
         remember_device: false,
       }
+      expect(SCrypt::Engine).to receive(:hash_secret).once.and_call_original
```

Comment:
For any auth attempt, we want to run an scrypt hash exactly once.

This and the line above are helpful because they catch both the case where we may hash multiple times and the case where we hash zero times.